File: Examples.GridDesktop/CSharp/GridDesktop.Examples/RunExamples.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using GridDesktop.Examples.WorkingWithGrid;
using GridDesktop.Examples.Articles;
using GridDesktop.Examples.WorkingWithWorksheet;
using GridDesktop.Examples.WorkingWithRowsandColumns;
using GridDesktop.Examples.WorkingWithCells;

namespace GridDesktop.Examples 
{
    public partial class RunExamples : Form
    {
        public RunExamples()
        {
            InitializeComponent();
        }

        private void closeAllToolStripMenuItem_Click(object sender, EventArgs e)
        {
            foreach (Form frm in this.MdiChildren)
            {
                if (frm != this)
                {
                    frm.Close();
                }
            }
        }

        private void exitToolStripMenuItem_Click(object sender, EventArgs e)
        {
            Application.Exit();
        }

        private void copyAndPasteRowsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            CopyAndPasteRows frmCopyPasteRows = new CopyAndPasteRows();
            frmCopyPasteRows.MdiParent = this;
            frmCopyPasteRows.Show();
        }

        private void dataBindingFeaturesInWorksheetsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            DataBindingFeature frmDataBindingFeatures = new DataBindingFeature();
            frmDataBindingFeatures.MdiParent = this;
            frmDataBindingFeatures.Show();
        }

        private void managingContextMenuToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingContextMenu frmManagingContextMenu = new ManagingContextMenu();
            frmManagingContextMenu.MdiParent = this;
            frmManagingContextMenu.Show();
        }

        private void openingExcelFileToolStripMenuItem_Click(object sender, EventArgs e)
        {
            OpeningExcelFile frmOpeningExcelFile = new OpeningExcelFile();
            frmOpeningExcelFile.MdiParent = this;
            frmOpeningExcelFile.Show();
        }

        private void savingExcelFileToolStripMenuItem_Click(object sender, EventArgs e)
        {
            SavingExcelFile frmSavingExcelFile = new SavingExcelFile();
            frmSavingExcelFile.MdiParent = this;
            frmSavingExcelFile.Show();
        }

        private void gridDesktopEventsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            GridDesktopEvents frmGridDesktopEvents = new GridDesktopEvents();
            frmGridDesktopEvents.MdiParent = this;
            frmGridDesktopEvents.Show();
        }

        private void accessingWorksheetsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AccessingWorksheets frmAccessingWorksheets = new AccessingWorksheets();
            frmAccessingWorksheets.MdiParent = this;
            frmAccessingWorksheets.Show();
        }

        private void addORInsertWorksheetsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddInsertWorksheet frmAddInsertWorksheet = new AddInsertWorksheet();
            frmAddInsertWorksheet.MdiParent = this;
            frmAddInsertWorksheet.Show();
        }

        private void removeAWorksheetToolStripMenuItem_Click(object sender, EventArgs e)
        {
            RemoveWorksheet frmRemoveWorksheet = new RemoveWorksheet();
            frmRemoveWorksheet.MdiParent = this;
            frmRemoveWorksheet.Show();
        }

        private void renameAWorksheetToolStripMenuItem_Click(object sender, EventArgs e)
        {
            RenameWorksheet frmRenameWorksheet = new RenameWorksheet();
            frmRenameWorksheet.MdiParent = this;
            frmRenameWorksheet.Show();
        }

        private void importDataFromDataTableToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ImportDataFromDataTable frmImportData = new ImportDataFromDataTable();
            frmImportData.MdiParent = this;
            frmImportData.Show();
        }

        private void exportDataToDataTableToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ExportDataToDataTable frmExportData = new ExportDataToDataTable();
            frmExportData.MdiParent = this;
            frmExportData.Show();
        }

        private void workingWithValidationsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ValidationInWorksheets frmValidations = new ValidationInWorksheets();
            frmValidations.MdiParent = this;
            frmValidations.Show();
        }

        private void sortDataToolStripMenuItem_Click(object sender, EventArgs e)
        {
            SortData frmSortData = new SortData();
            frmSortData.MdiParent = this;
            frmSortData.Show();
        }

        private void managingHyperlinksToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingHyperlinks frmHyperlinks = new ManagingHyperlinks();
            frmHyperlinks.MdiParent = this;
            frmHyperlinks.Show();
        }

        private void managingPicturesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingPictures frmManagingPictures = new ManagingPictures();
            frmManagingPictures.MdiParent = this;
            frmManagingPictures.Show();
        }

        private void managingCommentsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingComments frmManagingComments = new ManagingComments();
            frmManagingComments.MdiParent = this;
            frmManagingComments.Show();
        }

        private void addingCellControlsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddingCellControls frmAddingCellControls = new AddingCellControls();
            frmAddingCellControls.MdiParent = this;
            frmAddingCellControls.Show();
        }

        private void managingCellControlsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingCellControls frmManagingCellControls = new ManagingCellControls();
            frmManagingCellControls.MdiParent = this;
            frmManagingCellControls.Show();
        }

        private void showHideScrolBarsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            DisplayHideScrolBars frmDisplayHideScrols = new DisplayHideScrolBars();
            frmDisplayHideScrols.MdiParent = this;
            frmDisplayHideScrols.Show();
        }

        private void movingWorksheetsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            MovingWorksheets frmMovingWorksheets = new MovingWorksheets();
            frmMovingWorksheets.MdiParent = this;
            frmMovingWorksheets.Show();
        }

        private void readingDataValidationsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ReadingDataValidations frmDataValidations = new ReadingDataValidations();
            frmDataValidations.MdiParent = this;
            frmDataValidations.Show();
        }

        private void zoomingInOrOutToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ZoomingInOut frmZoomingInOut = new ZoomingInOut();
            frmZoomingInOut.MdiParent = this;
            frmZoomingInOut.Show();
        }

        private void addInsertColumnToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddInsertColumn frmAddInsertColumn = new AddInsertColumn();
            frmAddInsertColumn.MdiParent = this;
            frmAddInsertColumn.Show();
        }

        private void removingAColmnToolStripMenuItem_Click(object sender, EventArgs e)
        {
            RemovingColumn frmRemovingColumn = new RemovingColumn();
            frmRemovingColumn.MdiParent = this;
            frmRemovingColumn.Show();
        }

        private void addInsertRowToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddInsertRow frmAddInsertRow = new AddInsertRow();
            frmAddInsertRow.MdiParent = this;
            frmAddInsertRow.Show();
        }

        private void removingARowToolStripMenuItem_Click(object sender, EventArgs e)
        {
            RemovingRow frmRemovingRow = new RemovingRow();
            frmRemovingRow.MdiParent = this;
            frmRemovingRow.Show();
        }

        private void applyingStyleOnRowColumnToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ApplyingStyleOnRowColumn frmApplyStyle = new ApplyingStyleOnRowColumn();
            frmApplyStyle.MdiParent = this;
            frmApplyStyle.Show();
        }

        private void settingColumnWidhtRowHeightToolStripMenuItem_Click(object sender, EventArgs e)
        {
            SettingColumnWidthAndRowHeight frmSettingHeightWidth = new SettingColumnWidthAndRowHeight();
            frmSettingHeightWidth.MdiParent = this;
            frmSettingHeightWidth.Show();
        }

        private void freezeUnfreezeRowsColumnsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            FreezeUnfreezeRowsColumns frmFreezeUnfreeze = new FreezeUnfreezeRowsColumns();
            frmFreezeUnfreeze.MdiParent = this;
            frmFreezeUnfreeze.Show();
        }

        private void addingCellControlsInColumnsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddingCellControlsInColumns frmAddControl = new AddingCellControlsInColumns();
            frmAddControl.MdiParent = this;
            frmAddControl.Show();
        }

        private void workingWithColumnValidationsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            WorkingWithColumnValidations frmColValidations = new WorkingWithColumnValidations();
            frmColValidations.MdiParent = this;
            frmColValidations.Show();
        }

        private void managingControlsInColumnsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ManagingControlsInColumns frmManageControls = new ManagingControlsInColumns();
            frmManageControls.MdiParent = this;
            frmManageControls.Show();
        }

        private void changeFontColorOfRowColumnToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ChangeFontColorRowColumn frmChangeFontColor = new ChangeFontColorRowColumn();
            frmChangeFontColor.MdiParent = this;
            frmChangeFontColor.Show();
        }

        private void accessAndModifyCellsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AccessAndModifyCells frmAccessAndModify = new AccessAndModifyCells();
            frmAccessAndModify.MdiParent = this;
            frmAccessAndModify.Show();
        }

        private void accessingCellsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AccessingCells frmAccessCells = new AccessingCells();
            frmAccessCells.MdiParent = this;
            frmAccessCells.Show();
        }

        private void addCellProtectionToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddCellProtection frmAddCellProtection = new AddCellProtection();
            frmAddCellProtection.MdiParent = this;
            frmAddCellProtection.Show();
        }

        private void addingCellFormulasToolStripMenuItem_Click(object sender, EventArgs e)
        {
            AddingCellFormulas frmAddFormula = new AddingCellFormulas();
            frmAddFormula.MdiParent = this;
            frmAddFormula.Show();
        }

        private void applyingStyleOnCellsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ApplyStyleOnCells frmApplyStyle = new ApplyStyleOnCells();
            frmApplyStyle.MdiParent = this;
            frmApplyStyle.Show();
        }

        private void changeFontColorOfCellToolStripMenuItem_Click(object sender, EventArgs e)
        {
            ChangeFontColorOfCell frmChangeFontColor = new ChangeFontColorOfCell();
            frmChangeFontColor.MdiParent = this;
            frmChangeFontColor.Show();
        }

        private void filteringDataToolStripMenuItem_Click(object sender, EventArgs e)
        {
            FilteringData frmFilterData = new FilteringData();
            frmFilterData.MdiParent = this;
            frmFilterData.Show();
        }

        private void formattingCellRangeToolStripMenuItem_Click(object sender, EventArgs e)
        {
            FormattingCellRange frmFormattinCellRange = new FormattingCellRange();
            frmFormattinCellRange.MdiParent = this;
            frmFormattinCellRange.Show();
        }

        private void mergingAndUnmergingCellsToolStripMenuItem_Click(object sender, EventArgs e)
        {
            MergingAndUnMergingCells frmMergeUnmerge = new MergingAndUnMergingCells();
            frmMergeUnmerge.MdiParent = this;
            frmMergeUnmerge.Show();
        }

        private void undoAndRedoFeatureToolStripMenuItem_Click(object sender, EventArgs e)
        {
            UndoRedoFeature frmUndoRedo = new UndoRedoFeature();
            frmUndoRedo.MdiParent = this;
            frmUndoRedo.Show();
        }

        private void usingFormatPainterToolStripMenuItem_Click(object sender, EventArgs e)
        {
            UsingFormatPainter frmFormatPainter = new UsingFormatPainter();
            frmFormatPainter.MdiParent = this;
            frmFormatPainter.Show();
        }

        private void usingNamedRangesToolStripMenuItem_Click(object sender, EventArgs e)
        {
            UsingNamedRanges frmUsingNamedRanges = new UsingNamedRanges();
            frmUsingNamedRanges.MdiParent = this;
            frmUsingNamedRanges.Show();
        }
    }
}

File: Examples.GridDesktop/VisualBasic/GridDesktop.Examples/RunExamples.vb
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports GridDesktop.Examples.WorkingWithGrid
Imports GridDesktop.Examples.Articles
Imports GridDesktop.Examples.WorkingWithWorksheet
Imports GridDesktop.Examples.WorkingWithRowsandColumns

Public Partial Class RunExamples
	Inherits Form
	Public Sub New()
		InitializeComponent()
	End Sub

	Private Sub closeAllToolStripMenuItem_Click(sender As Object, e As EventArgs)
		For Each frm As Form In Me.MdiChildren
			If frm IsNot Me Then
				frm.Close()
			End If
		Next
	End Sub

	Private Sub exitToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Application.[Exit]()
	End Sub

	Private Sub copyAndPasteRowsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmCopyPasteRows As New CopyAndPasteRows()
		frmCopyPasteRows.MdiParent = Me
		frmCopyPasteRows.Show()
	End Sub

	Private Sub dataBindingFeaturesInWorksheetsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmDataBindingFeatures As New DataBindingFeature()
		frmDataBindingFeatures.MdiParent = Me
		frmDataBindingFeatures.Show()
	End Sub

	Private Sub managingContextMenuToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmManagingContextMenu As New ManagingContextMenu()
		frmManagingContextMenu.MdiParent = Me
		frmManagingContextMenu.Show()
	End Sub

	Private Sub openingExcelFileToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmOpeningExcelFile As New OpeningExcelFile()
		frmOpeningExcelFile.MdiParent = Me
		frmOpeningExcelFile.Show()
	End Sub

	Private Sub savingExcelFileToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmSavingExcelFile As New SavingExcelFile()
		frmSavingExcelFile.MdiParent = Me
		frmSavingExcelFile.Show()
	End Sub

	Private Sub gridDesktopEventsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmGridDesktopEvents As New GridDesktopEvents()
		frmGridDesktopEvents.MdiParent = Me
		frmGridDesktopEvents.Show()
	End Sub

	Private Sub accessingWorksheetsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAccessingWorksheets As New AccessingWorksheets()
		frmAccessingWorksheets.MdiParent = Me
		frmAccessingWorksheets.Show()
	End Sub

	Private Sub addORInsertWorksheetsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAddInsertWorksheet As New AddInsertWorksheet()
		frmAddInsertWorksheet.MdiParent = Me
		frmAddInsertWorksheet.Show()
	End Sub

	Private Sub removeAWorksheetToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmRemoveWorksheet As New RemoveWorksheet()
		frmRemoveWorksheet.MdiParent = Me
		frmRemoveWorksheet.Show()
	End Sub

	Private Sub renameAWorksheetToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmRenameWorksheet As New RenameWorksheet()
		frmRenameWorksheet.MdiParent = Me
		frmRenameWorksheet.Show()
	End Sub

	Private Sub importDataFromDataTableToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmImportData As New ImportDataFromDataTable()
		frmImportData.MdiParent = Me
		frmImportData.Show()
	End Sub

	Private Sub exportDataToDataTableToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmExportData As New ExportDataToDataTable()
		frmExportData.MdiParent = Me
		frmExportData.Show()
	End Sub

	Private Sub workingWithValidationsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmValidations As New ValidationInWorksheets()
		frmValidations.MdiParent = Me
		frmValidations.Show()
	End Sub

	Private Sub sortDataToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmSortData As New SortData()
		frmSortData.MdiParent = Me
		frmSortData.Show()
	End Sub

	Private Sub managingHyperlinksToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmHyperlinks As New ManagingHyperlinks()
		frmHyperlinks.MdiParent = Me
		frmHyperlinks.Show()
	End Sub

	Private Sub managingPicturesToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmManagingPictures As New ManagingPictures()
		frmManagingPictures.MdiParent = Me
		frmManagingPictures.Show()
	End Sub

	Private Sub managingCommentsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmManagingComments As New ManagingComments()
		frmManagingComments.MdiParent = Me
		frmManagingComments.Show()
	End Sub

	Private Sub addingCellControlsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAddingCellControls As New AddingCellControls()
		frmAddingCellControls.MdiParent = Me
		frmAddingCellControls.Show()
	End Sub

	Private Sub managingCellControlsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmManagingCellControls As New ManagingCellControls()
		frmManagingCellControls.MdiParent = Me
		frmManagingCellControls.Show()
	End Sub

	Private Sub showHideScrolBarsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmDisplayHideScrols As New DisplayHideScrolBars()
		frmDisplayHideScrols.MdiParent = Me
		frmDisplayHideScrols.Show()
	End Sub

	Private Sub movingWorksheetsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmMovingWorksheets As New MovingWorksheets()
		frmMovingWorksheets.MdiParent = Me
		frmMovingWorksheets.Show()
	End Sub

	Private Sub readingDataValidationsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmDataValidations As New ReadingDataValidations()
		frmDataValidations.MdiParent = Me
		frmDataValidations.Show()
	End Sub

	Private Sub zoomingInOrOutToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmZoomingInOut As New ZoomingInOut()
		frmZoomingInOut.MdiParent = Me
		frmZoomingInOut.Show()
	End Sub

	Private Sub addInsertColumnToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAddInsertColumn As New AddInsertColumn()
		frmAddInsertColumn.MdiParent = Me
		frmAddInsertColumn.Show()
	End Sub

	Private Sub removingAColmnToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmRemovingColumn As New RemovingColumn()
		frmRemovingColumn.MdiParent = Me
		frmRemovingColumn.Show()
	End Sub

	Private Sub addInsertRowToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAddInsertRow As New AddInsertRow()
		frmAddInsertRow.MdiParent = Me
		frmAddInsertRow.Show()
	End Sub

	Private Sub removingARowToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmRemovingRow As New RemovingRow()
		frmRemovingRow.MdiParent = Me
		frmRemovingRow.Show()
	End Sub

	Private Sub applyingStyleOnRowColumnToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmApplyStyle As New ApplyingStyleOnRowColumn()
		frmApplyStyle.MdiParent = Me
		frmApplyStyle.Show()
	End Sub

	Private Sub settingColumnWidhtRowHeightToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmSettingHeightWidth As New SettingColumnWidthAndRowHeight()
		frmSettingHeightWidth.MdiParent = Me
		frmSettingHeightWidth.Show()
	End Sub

	Private Sub freezeUnfreezeRowsColumnsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmFreezeUnfreeze As New FreezeUnfreezeRowsColumns()
		frmFreezeUnfreeze.MdiParent = Me
		frmFreezeUnfreeze.Show()
	End Sub

	Private Sub addingCellControlsInColumnsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmAddControl As New AddingCellControlsInColumns()
		frmAddControl.MdiParent = Me
		frmAddControl.Show()
	End Sub

	Private Sub workingWithColumnValidationsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmColValidations As New WorkingWithColumnValidations()
		frmColValidations.MdiParent = Me
		frmColValidations.Show()
	End Sub

	Private Sub managingControlsInColumnsToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmManageControls As New ManagingControlsInColumns()
		frmManageControls.MdiParent = Me
		frmManageControls.Show()
	End Sub

	Private Sub changeFontColorOfRowColumnToolStripMenuItem_Click(sender As Object, e As EventArgs)
		Dim frmChangeFontColor As New ChangeFontColorRowColumn()
		frmChangeFontColor.MdiParent = Me
		frmChangeFontColor.Show()
	End Sub
End Class
